soccer/VisionReceiver.cpp

```diff
@@ -9,15 +9,18 @@
 
 using namespace std;
 
-VisionReceiver::VisionReceiver(bool sim, int port) {
-    simulation = sim;
-    _running = false;
-    this->port = port;
-}
+VisionReceiver::VisionReceiver(bool sim, int port)
+    : simulation(sim), _running(false), port(port) {}
 
 void VisionReceiver::stop() {
-    _running = false;
-    wait();
+    if (isRunning()) {
+        // Handle Case where start has been called, but _running has not been
+        // set to true yet.
+        while (!_running) {
```

Comment:
Ah I think I get it now. I don't think theres a cleaner solution w/o mutexes so I guess this is ok...